tendrl/commons/flows/unmanage_cluster/__init__.py

```diff
@@ -14,59 +21,203 @@ def run(self):
         ).load()
         if _cluster.is_managed == "no":
             if _cluster.current_job['job_name'] == self.__class__.__name__ \
-                and _cluster.current_job['status'] == 'finished':
+                    and _cluster.current_job['status'] == 'finished':
+                raise FlowExecutionFailedError(
+                    "Cluster is already in un-managed state"
+                )
+        if (_cluster.status is not None and
```

Comment:
I feel this check should not be done on child task of un-manage flow, so cant you move this inside if condition at L#55?

Reply:
Ack